src/util/dataAccess.cpp

```diff
@@ -165,21 +165,114 @@ ndsize_t positionToIndex(double position, const string &unit, const RangeDimensi
 }
 
 
+void getMaxExtent(const Dimension &dim, ndsize_t max_index, double &pos, double &ext) {
+    DimensionType dt = dim.dimensionType();
+    if (dt == DimensionType::Sample) {
+        SampledDimension sd = dim.asSampledDimension();
+        pos = sd.positionAt(0);
+        ext = sd.positionAt(max_index);
+    } else if (dt == DimensionType::Range) {
+        RangeDimension rd = dim.asRangeDimension();
+        pos = rd.tickAt(0);
+        ext = rd.tickAt(max_index);
+    } else if (dt == DimensionType::Set) {
+        SetDimension sd = dim.asSetDimension();
+        pos = 0.0;
+        if (max_index > pow(FLT_RADIX, std::numeric_limits<double>::digits)) {
+            throw nix::OutOfBounds("dataAccess::fillPositionsExtents: shape cannot be cast to double without loss of precision. Please open an issue on github!");
```

Comment:
This took me a bit, but maybe for similarly incapable floating point and math "dummies" as I am, a comment would be nice. 
Also, I think we can indeed make this a compile time constant by doing `(1 << std::numeric_limits<double>::digits)`